bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorageService.java

```diff
@@ -98,8 +102,13 @@ protected void deactivate() {
 
     @Override
     public <T> Storage<T> getStorage(String name, ClassLoader classLoader) {
-        File file = new File(dbFolderName, name + ".json");
+        File legacyFile = new File(dbFolderName, name + ".json");
+        File escapedFile = new File(dbFolderName, urlEscapeUnwantedChars(name) + ".json");
 
+        File file = escapedFile;
+        if (legacyFile.exists()) {
+            file = legacyFile;
+        }
```

Comment:
Instead of create two local file objects that are available in that scope you could outsource the logic to find the correct file object...

```Java
private File getStorageFile(final String name) {
    final File legacyFile = new File(dbFolderName, name + ".json");
    if (legacyFile.exists()) {
        return legacyFile;
    } else {
        return new File(dbFolderName, urlEscapeUnwantedChars(name) + ".json");
    }
}
```

Reply:
Hm, thinking about it again, it does perhaps not make sense to create a function if it is used only on that place.
No need to change it...